### PR TITLE
feat(aw_pose_covariance_modifier_node): add aw_pose_covariance_modifier for using NDT and GNSS together

### PR DESCRIPTION
**Pull request by meliketanrikulu**
## Description

This PR was created to use NDT and GNSS together in localization. In order for EKF to work properly, the covariance values of the given inputs must be provided correctly as input. In Default AW, NDT covariance values are provided with default values. GNSS, on the other hand, provides its own error in a healthy way. 

This PR also suggests that we drive with reference to the trusted source (GNSS), which can reliably calculate the error. It aims to drive with GNSS alone when the GNSS error is low, to use NDT-GNSS together when the GNSS error increases, and to use only NDT alone when the GNSS error goes beyond acceptable limits.

## Related links
This PR is related this proposal -> https://github.com/orgs/autowarefoundation/discussions/4134#discussioncomment-8394375

## Tests performed

Test results will be shared. 
## Notes for reviewers
This PR is currently shared as a draft.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [ ] I've confirmed the [contribution guidelines].
- [ ] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by KYabuuchi** [comment]
Thank you for creating the PR.  I really appreciate your contribution. :smile: 

From our previous discussions, I think it was decided to use the set_parameter service to modify the covariance of NDT. However, this PR is adding new service and new subscriber to NDT. Is there a specific reason for this change in architecture?

NDT already has a huge amount of feature, and  I think it would be preferable to perform calculations such as covariance modification within aw_pose_covariance_modifier.

**Comment by xmfcx** [comment]
Rebased to the latest main branch to make sure CI is running correctly.

File: localization/ndt_scan_matcher/CMakeLists.txt [hunk 1]
```diff
@@ -2,6 +2,7 @@ cmake_minimum_required(VERSION 3.14)
 project(ndt_scan_matcher)
 
 find_package(autoware_cmake REQUIRED)
+find_package(rcl_interfaces REQUIRED)
```
Comment (xmfcx) [inline localization/ndt_scan_matcher/CMakeLists.txt:5]:
```suggestion
```

**Comment by xmfcx** [comment]
This PR should be updated while considering these:
- NDT nodes shouldn't be changed unless it's absolutely necessary
- The additional logic should stay outside of existing nodes for ease of modularity
- The new additions shouldn't break existing structure

**Comment by meliketanrikulu** [comment]
> This PR should be updated while considering these:
> 
> * NDT nodes shouldn't be changed unless it's absolutely necessary
> * The additional logic should stay outside of existing nodes for ease of modularity
> * The new additions shouldn't break existing structure

Thank you @xmfcx .I will update the PR taking these into consideration. I shared the required architecture for this PR below the proposal. (https://github.com/orgs/autowarefoundation/discussions/4134#discussioncomment-8693020)

**Comment by xmfcx** [comment]
@meliketanrikulu I'm closing this PR.
Could you open the updated proposal in a new PR?

**Pull request closed without merging**